pkg/codegen/importer/hcl2.go

```diff
@@ -0,0 +1,467 @@
+// Copyright 2016-2020, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package importer
+
+import (
+	"fmt"
+	"math"
+	"strings"
+
+	"github.com/pulumi/pulumi/pkg/v2/codegen/hcl2/model"
+	"github.com/pulumi/pulumi/pkg/v2/codegen/hcl2/syntax"
+	"github.com/pulumi/pulumi/pkg/v2/codegen/schema"
+	"github.com/pulumi/pulumi/pkg/v2/resource/deploy/providers"
+	"github.com/pulumi/pulumi/sdk/v2/go/common/resource"
+	"github.com/pulumi/pulumi/sdk/v2/go/common/util/contract"
+	"github.com/zclconf/go-cty/cty"
+)
+
+// Null represents Pulumi HCL2's `null` variable.
+var Null = &model.Variable{
+	Name:         "null",
+	VariableType: model.NoneType,
+}
+
+// GenerateHCL2Definition generates a Pulumi HCL2 definition for a given resource.
+func GenerateHCL2Definition(loader schema.Loader, state *resource.State, names NameTable) (*model.Block, error) {
+	// TODO: pull the package version from the resource's provider
+	pkg, err := loader.LoadPackage(string(state.Type.Package()), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	r, ok := pkg.GetResource(string(state.Type))
+	if !ok {
+		return nil, fmt.Errorf("unknown resource type '%v'", r)
+	}
+
+	var items []model.BodyItem
+	for _, p := range r.InputProperties {
+		x, err := generatePropertyValue(p, state.Inputs[resource.PropertyKey(p.Name)])
+		if err != nil {
+			return nil, err
+		}
+		if x != nil {
+			items = append(items, &model.Attribute{
+				Name:  p.Name,
+				Value: x,
+			})
+		}
+	}
+
+	resourceOptions, err := makeResourceOptions(state, names)
+	if err != nil {
+		return nil, err
+	}
+	if resourceOptions != nil {
+		items = append(items, resourceOptions)
+	}
+
+	typ, name := state.URN.Type(), state.URN.Name()
+	return &model.Block{
+		Tokens: syntax.NewBlockTokens("resource", string(name), string(typ)),
```

Comment:
```suggestion
		Tokens: syntax.NewBlockTokens("resource", name, string(typ)),
```
`name` is already a `string`

Reply:
`name` is a `tokens.QName`: https://pkg.go.dev/github.com/pulumi/pulumi/sdk@v1.14.1/go/common/resource?tab=doc#URN.Name

Reply:
Huh, I guess the sourcegraph hint was wrong then.